### PR TITLE
sidecar: testcase should not add service entry bind to loopback address

### PR DESCRIPTION
**Pull request by silentdai**
Find TestSidecarScope failure when switch inbound listener to filter chain in PR #16145

Previously in the testcase we use 127.255.X.Y to simulate the dns resolved ip of an service entry host and assume the request will go through envoy outbound listener.

When I split outbound and inbound listener, the rule is that 
1. non 127.0.0.1 will go through inbound virtual listener and,
2. inbound virtual listener doesn't hand off the connection to a not-bind listener on 127.255.X.Y:<PortOfServiceEntry>

The TestSidecarScope would fail to reach the endpoint defined in service entry regardless it is private or public in sidecar scope.

This PR use 192.0.2.X as the ip address in service entry and sidecar.  These addresses should never be assigned to any entity and iptables rules would consider them as non-local address.

The traffic to those host will be going through outbound listener as assumed in the test case. The test case would pass regardless the inbound virtual listener uses filter chain or not.

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/e2e/tests/pilot/testdata/networking/v1alpha3/service-entry-tcp-scope-public.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ spec:
   hosts:
   - "service.tcp.com"
   addresses:
-  - "127.255.255.11"
+  - "192.168.9.251"
```
Comment (rshriram) [inline tests/e2e/tests/pilot/testdata/networking/v1alpha3/service-entry-tcp-scope-public.yaml]: Dont change this.. We intentionally use the 127.255.255.* IPs for various reasons (multicluster, or to ensure that this IP is not colliding with anything else on the network, and this is also tied in with the coreDNS thing we have).

**Comment by rshriram** [comment]
So instead of changing the IP, is it possible to change the Sidecar to use a `captureMode: NONE` so that the inbound listener continues to function on that port/IP ?

**Comment by silentdai** [comment]
@rshriram  I understand the concern that the ip should not collide with any existing system. 127.X.Y.Z is a good choice. 
I think captureMode:NONE should be another test in additional to the existing one...

What do you think if we use a well known ip instead?  It should work since this IP as destination is expected to capture and redirected by sidecar proxy. The test case never connects to this ip.

**Comment by silentdai** [comment]
Use 192.0.2.0/24 for service entries defined in test. 
These ip are not assigned to any entity. These IPs are supposed to captured by iptables.
The tests are supposed to pass consistently.
https://tools.ietf.org/html/rfc5737

**Comment by silentdai** [comment]
@rshriram  gentle ping

File: tests/e2e/tests/pilot/testdata/networking/v1alpha3/service-entry-tcp-scope-public.yaml (hunk 1 above)
Comment (silentdai) [reply tests/e2e/tests/pilot/testdata/networking/v1alpha3/service-entry-tcp-scope-public.yaml]: changed to 192.0.2.X

**Comment by lambdai** [comment]
/cherrypick release-1.3

CC @rlenglet  As dependency of back porting #16414

**Pull request merged**